### ISSUE TITLE
Mixing single and double quotes in Enum causes false syntax error

### ISSUE THREAD

**Issue by sambarluc** [PyCQA/pylint#2676]
### Steps to reproduce
This can be reproduced by these lines:
```
from enum import Enum
class A(Enum):
    a = 'x"y"'

```

### Current behavior

Pylint currently returns `5:0: E0001: invalid syntax (<unknown>, line 5) (syntax-error)`

Interestingly, if I change the script to:
```
from enum import Enum


class A(Enum):
    a = 'a"b"'

```

then pylint returns: `5:0: E0001: cannot mix bytes and nonbytes literals (<unknown>, line 5) (syntax-error)`

To me this strongly suggests there is some issue in parsing the double quotes.

### Expected behavior

This is valid python code which doesn't raise any exception when run by the interpreter.

### pylint --version output

```
$ pylint --version
pylint 2.2.2
astroid 2.1.0
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]

```

**Comment by sambarluc** [comment]
I forgot to add that this only happens when creating an `Enum`. No problem if the class inherits from nothing/`object`/something else.

**Comment by anikey-m** [comment]
Got same problem for enum with strings values that contain "special" characters.
```
import enum
class Example(enum.Enum):
    NULL = '\N{NULL}'
```
Pylint report error `1:0: E0001: source code string cannot contain null bytes (syntax-error)`.

Also found another errors:
* `E0001: unexpected indent (<unknown>, line 2) (syntax-error)` - for `'\N{END OF LINE}'`
* `E0001: EOL while scanning string literal (<unknown>, line 5) (syntax-error)` - for `'\N{CARRIAGE RETURN}'`

No problem when class not inherited from `Enum`. Think there is some problem with enum values processing.

**pylint --version output**
```
pylint 2.2.2
astroid 2.1.0
Python 3.7.2 (default, Dec 29 2018, 21:15:15) 
[GCC 8.2.1 20181127]
```

**Comment by PCManticore** [comment]
Thanks for reporting a bug, should be fixed in astroid's master.
